controllers/nutanixmachine_controller.go

```diff
@@ -513,8 +513,8 @@ func (r *NutanixMachineReconciler) reconcileNode(rctx *nctx.MachineContext) (rec
 }
 
 func (r *NutanixMachineReconciler) validateMachineConfig(rctx *nctx.MachineContext) error {
-	if len(rctx.NutanixMachine.Spec.Subnets) == 0 {
-		return fmt.Errorf("atleast one subnet is needed to create the VM %s", rctx.NutanixMachine.Name)
+	if rctx.Machine.Spec.FailureDomain == nil && len(rctx.NutanixMachine.Spec.Subnets) == 0 {
```

Comment:
given your above comment that we need to pass this for backward compatibility only. what if i dont set this but set fd directly, this check will not let it go ahead rite so this make it mandatory>?

Reply:
Updated the check